security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -78,14 +82,27 @@ func (c *CitadelClient) Close() {
 
 // CSR Sign calls Citadel to sign a CSR.
 func (c *CitadelClient) CSRSign(csrPEM []byte, certValidTTLInSec int64) ([]string, error) {
+	crMetadata := map[string]interface{}{
+		security.CertSigner: c.opts.CertSigner,
+	}
+	crMetaStruct, err := mapToStruct(crMetadata)
```

Comment:
very nit: you can just construct the Struct directly? to avoid some complexity and error handling

Reply:
Thanks for the great suggestion!